### PR TITLE
Evaluate testtrain mode

### PR DESCRIPTION
**Pull request by rgstephens**
**Proposed changes**:
- A new evaluate mode that dynamically creates a test set from the training data set and then runs an evaluate. Discussed in issue #1625. Example syntax:

```
python rasa_nlu.evaluate.py --mode traintest --config nlu_config.yml --data training/ --pct 90 --cutoff 5
```

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by akelad** [comment]
Thanks for creating this PR, we'll give it a review as soon as possible

**Comment by tmbo** [comment]
This one went stale, sorry for that. Files have moved on the master branch, so we can't merge this one anymore.
